File: SocialMediaApi/models/post.cs
using System;
using System.Collections.Generic;

namespace SocialMediaApi.Models
{
    public class Post
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public required string ImageUrl { get; set; }
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public int LikesCount { get; set; } = 0;
        public required ApplicationUser User { get; set; }
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<Like> Likes { get; set; } = new List<Like>();
    }
}

File: subReact/frontend/src/types/post.ts
import type { UserInfo } from "~/types/user";
import type { Comment } from "~/types/comment";
import type { Like } from "~/types/like";
export interface Post {
  id: number;
  title: string;
  imageUrl: string;
  created: string;
  user: UserInfo;
  comments: Comment[];
  likesCount: number;
  likes: Like[];
}
export interface PostData {
  title: string;
  image: File;
}
